src/ChannelServer/World/Entities/Components/Buff.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using Melia.Channel.Buffs.Base;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.Data.Database;
+
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Buff
+	/// </summary>
+	public class Buff : IUpdateable
+	{
+		/// <summary>
+		/// The buff's owner.
+		/// </summary>
+		public ICombatEntity Caster { get; }
+
+		/// <summary>
+		/// The buff's target.
+		/// </summary>
+		public ICombatEntity Target { get; }
+
+		/// <summary>
+		/// Returns the buff's id.
+		/// </summary>
+		public BuffId Id { get; }
+
+		/// <summary>
+		/// Returns the buff's associated skill id.
+		/// Default value is 1.
+		/// </summary>
+		public SkillId SkillId { get; } = SkillId.Normal_Attack;
+
+		/// <summary>
+		/// The buff's data from the buff database.
+		/// </summary>
+		public BuffData Data { get; }
+
+		/// <summary>
+		/// The buff's duration, separate from database duration
+		/// because database duration is inaccurate.
+		/// </summary>
+		public int Duration { get; }
+
+		/// <summary>
+		/// Index in world collection?
+		/// </summary>
+		public int Handle { get; private set; }
+
+		/// <summary>
+		/// Returns the buff's overbuff count. If this value reaches the buff's
+		/// maximum overbuff, the buff has additional effects based on this
+		/// counter.
+		/// </summary>
+		public int OverbuffCounter { get; private set; }
+
+		/// <summary>
+		/// The buff's removal time
+		/// </summary>
+		public DateTime RemovalTime { get; set; }
+
+		/// <summary>
+		/// If the buff has an update time
+		/// </summary>
+		/// <returns></returns>
+		public bool HasUpdateTime => this.Data.UpdateTime != 0;
+
+		/// <summary>
+		/// The buff's next update time
+		/// </summary>
+		public DateTime NextUpdateTime { get; set; }
+
+		/// <summary>
+		/// The buff's behavior handler
+		/// </summary>
+		public IBuffHandler Handler { get; private set; }
+
+		/// <summary>
+		/// Creates a new instance.
+		/// </summary>
+		/// <param name="caster"></param>
+		/// <param name="target"></param>
+		/// <param name="buffId"></param>
+		/// <param name="duration"></param>
+		/// <param name="skillId"></param>
+		public Buff(ICombatEntity caster, ICombatEntity target, BuffId buffId, int duration = 0, SkillId skillId = SkillId.Normal_Attack)
```

Comment:
What do you think about using `TimeSpan` instead of `int` for the duration? It's a little more to type, but it makes it much clearer what you're working with. A simple integer, especially if you didn't document anything about it, could be anything. Ticks, seconds, minutes, you'd have to guess or look up other uses of the method.

Reply:
I don't mind it, but I haven't used TimeSpan before, even after looking up the documents the initializer can use ticks or hours, minutes, seconds. So I'm not too sure how about to sure how to implement this as I'm using milliseconds right now. I've renamed the parameter to be more verbose durationInMilliseconds.

Reply:
For the parameter you would just use the type `TimeSpan` and when creating a buff you would, for example, use `TimeSpan.FromMilliseconds(1000)` to create it. For a time I disliked this, because of how verbose it is, but the clarity is unbeatable, which comes in handy in bigger projects that various people work on, because there's always cases where seconds or something might make more sense than milliseconds, and suddenly you have different types of durations in different parts of the code. With TimeSpan und you can just use different types when it suits you, like `TimeSpan.FromMinutes(1)`.